pypeit/scienceimage.py

```diff
@@ -119,23 +75,19 @@ class ScienceImage(processimages.ProcessImages):
     frametype = 'science'
 
     # TODO: Merge into a single parset, one for procing, and one for scienceimage
-    def __init__(self, tslits_dict, spectrograph, file_list, bg_file_list = [], ir_redux=False,
-                 det=1, objtype='science', binning = None, setup=None,
-                 par=None, frame_par=None):
+    def __init__(self, spectrograph, file_list, bg_file_list = [], ir_redux=False, det=1, objtype='science',
```

Comment:
I would make `par` a required input

Reply:
I agree that `self.par` need not be required input but
`self.par['process']` should be instead.

All primary objects should have `par` passed in
(or just the piece needed).

And I agree about `objtype`

Reply:
All of this `par` pieces should be required arguments, not optional.
And will be eventually.  I'll take the blame for the other base classes
(and will modify them soon enough).

Reply:
I don't think the parset should be required if spectrograph is required, since you can easily get the default parset.  This then requires fewer incantations to get going.